modules/@apostrophecms/rich-text-widget/index.js

```diff
@@ -675,6 +778,59 @@ module.exports = {
           }
         }
         return content;
+      },
+      // Validate the types provided for links, update labels derived from
+      // corresponding modules, as they are not available at the time
+      // the schema is generated.
+      validateAndFixLinkWithTypes() {
+        const linkWithType = (Array.isArray(self.options.linkWithType)
+          ? self.options.linkWithType
+          : [ self.options.linkWithType ]);
+
+        for (const type of linkWithType) {
+          if (!self.apos.modules[type]) {
+            throw new Error(
+              `The linkWithType option of rich text widget "${type}" must be a valid module type`
+            );
+          }
+
+          self.linkFields[`_${type}`].label = getLabel(type);
+          const choice = self.linkFields.linkTo.choices
+            .find(choice => choice.value === type);
+
+          choice.label = getLabel(type);
+        }
+
+        function getLabel(type) {
+          if ([ '@apostrophecms/any-page-type', '@apostrophecms/page' ].includes(type)) {
+            return 'apostrophe:page';
+          }
+          return self.apos.modules[type].options?.label ?? type;
+        }
+      },
+      // Compose and register the link schema.
+      composeLinkSchema() {
+        self.linkSchema = self.apos.schema.compose({
+          addFields: self.apos.schema.fieldsToArray(`Links ${self.__meta.name}`, self.linkFields),
+          arrangeFields: self.apos.schema.groupsToArray(self.linkFieldsGroups)
+        }, self);
+        const forbiddenFields = [
+          '_id'
+        ];
+        self.linkSchema.forEach((field) => {
+          if (forbiddenFields.includes(field.name)) {
+            throw new Error('Doc type ' + self.name + ': the field name ' + field.name + ' is forbidden');
+          }
+        });
+        self.apos.schema.validate(self.linkSchema, {
+          type: 'doc type',
```

Comment:
You can set type to 'link' here. It's debugging info to provide context for error messages, and "doc type" means an apostrophe document.